register.go

```diff
@@ -0,0 +1,91 @@
+package currency
+
+import (
+	"fmt"
+)
+
+// EmptyCurrencyCodeError indicates that the currency code was empty.
+type EmptyCurrencyCodeError struct{}
+
+func (e EmptyCurrencyCodeError) Error() string {
+	return "register currency error: empty currency code"
+}
+
+// CurrencyAlreadyExistsError indicates that the currency code already exists in the ISO list.
+type CurrencyAlreadyExistsError struct {
+	Code string
+}
+
+func (e CurrencyAlreadyExistsError) Error() string {
+	return fmt.Sprintf("register currency error: code %q already exists in ISO list", e.Code)
+}
+
+// RegisterCurrencyOptions defines parameters for registering a new or custom currency.
+type RegisterCurrencyOptions struct {
+	// NumericCode is usually a three-digit code, for example "999".
+	NumericCode string
+
+	// Digits is the number of decimal fraction digits.
+	Digits uint8
+
+	// SymbolData is a list of possible symbols and the locales
+	// in which each symbol is used.
+	//
+	// Example:
+	//    []SymbolData{
+	//       {Symbol: "₿", Locales: []string{"en"}},
+	//       {Symbol: "BTC", Locales: []string{"uk"}},
+	//    }
+	SymbolData []SymbolData
```

Comment:
It feels odd to call this SymbolData. Why not just Symbols?

I would expect to see `Symbols map[string]currency.Locale`, used like this:
```
Symbols: map[string][]currency.Locale{
			"₿": {
				currency.Locale{Language: "en"},
			},
			"BTC": {
				currency.Locale{Language: "uk"}},
			}
		},
```

That would make it match the GetSymbol API:
```
// GetSymbol returns the symbol for a currency code.
func GetSymbol(currencyCode string, locale Locale) (symbol string, ok bool) {
```

Having to initialize a currency.Locale is a bit annoying, but that's how all other parts of this package currently accept a locale, so we're locked in.

As a bonus, use of the map means you no longer need a separate struct for the symbol data.